.github/workflows/push_stata_licensed_image.yml

```diff
@@ -0,0 +1,72 @@
+name: Push Stata Image to quay
+
+on:
+  push:
+    paths:
+      - jupyter-pystata-licensed/**
+      - .github/workflows/push_stata_licensed_image.yml
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Maximize build space
+        uses: easimon/maximize-build-space@master
+        with:
+          root-reserve-mb: 30000
+          swap-size-mb: 1024
+          remove-dotnet: 'true'
+          remove-android: 'true'
+          remove-haskell: 'true'
+      - uses: actions/checkout@v2
+      - uses: prewk/s3-cp-action@v0.1.1
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          SOURCE: 's3://ctds-stata/Stata17Linux64.tar.gz'
+          DEST: './jupyter-pystata-licensed/resources/'
+
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=$(echo $(echo ${GITHUB_REF#refs/heads/} | tr  / _))" >> $GITHUB_OUTPUT
+        id: extract_branch
+
+      - name: Determine image to build
+        id: parse_image
+        shell: python
+        run: |
+          import os
+
+          build_target = "jupyter-pystata-licensed"
+          print(f"Will trigger build for: {build_target}")
+          with open(os.environ['GITHUB_OUTPUT'], 'a') as fh:
+            print(f'build_target={build_target}', file=fh)
+
+      - if: ${{ steps.parse_image.outputs.build_target }}
+        name: Sanitize image name
+        id: sanitize_name
+        run: |
+          IMAGE_NAME=$( sed 's/[^[:alnum:]]/_/g' <<< ${{ steps.parse_image.outputs.build_target }} );
+          echo "image_name=$IMAGE_NAME" >> $GITHUB_OUTPUT
+
+      - name: Build Image
+        id: build-image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: stata-heal
```

Comment:
or "jupyter-pystata-licenced" and "jupyter-pystata-byol" if you want to keep the same convention in the folders in this repo

Reply:
Could we use "jupyter-pystata-gen3-licensed" and "jupyter-pystata-user-licensed"? This would be consistent with the workspace naming in the hatchery config.

Reply:
sure

Reply:
The gen3-licensed workspace will also require and update to the distribute-licenses-job.yml in the cloud-auto repo. We will need to change "stata-heal" to the new name "jupyter-pystata-gen3-licensed" in the [line for searching for container images.](https://github.com/uc-cdis/cloud-automation/blob/2bed77c34ae7a0830b8b92d9238d8f892494f306/kube/services/jobs/distribute-licenses-job.yaml#L79). I will create a PR for that change.